compositor_render/src/scene/web_view_component.rs

```diff
@@ -0,0 +1,62 @@
+use std::sync::Arc;
+
+use crate::transformations::web_renderer::WebRenderer;
+
+use super::{
+    scene_state::BuildStateTreeCtx, BuildSceneError, Component, ComponentId, IntermediateNode,
+    Size, StatefulComponent, WebViewComponent,
+};
+
+#[derive(Debug, Clone)]
+pub(super) struct StatefulWebViewComponent {
+    pub(super) id: Option<ComponentId>,
+    pub(super) children: Vec<StatefulComponent>,
+    pub(super) instance: Arc<WebRenderer>,
+}
```

Comment:
> What does Stateful refer to in the case of WebViewComponent? 

We have a Component enum that has a variant  e.g. ViewComponent. This is our public API.
We have a StatefulComponent enum that represents a component tree where each component can have a state. 

We need to represent each Component as StatefulComponent, so even if we do not need any custom state we need to have a type like StatefulWebViewComponent. In this case I would argue that state is useful because we hold on to Arc<WebRenderer> instance, so if it's unregistered we can still use it e.g. during end transition.

> Isn't the state of the web renderer internal to the instance?

It is.